modern/src/initialize.js

```diff
@@ -315,11 +341,11 @@ async function applyGroupDefs(root) {
   });
 }
 
-async function initialize(zip, skinXml) {
+async function initialize(zip, skinXml, sharedMakiTree = null) {
```

Comment:
We can split it out, but like you say, when we apply groupdefs we will need to have the shared tree, so Im not quite sure what that will save us.

Not sure this will help at all but:
1) When we initialize the freeform plugin, there are no additional shared components it needs to know about. So it is the skinXml, and there is no shared maki tree.
2) When we initialize the actual skin, we have the shared plugin converted to maki tree that it needs to know about